net-utils/src/sockets.rs

```diff
@@ -0,0 +1,56 @@
+use {
+    crate::{bind_common_in_range_with_config, bind_common_with_config, PortRange, SocketConfig},
+    std::{
+        net::{IpAddr, SocketAddr, TcpListener, UdpSocket},
+        sync::atomic::{AtomicU16, Ordering},
+    },
+};
+
+const BASE_PORT: u16 = 5000;
+
+/// Retrieve a free 100-port slice for unit tests
+///
+/// When running under nextest, this will try to provide
+/// a unique slice of port numbers (assuming no other nextest processes
+/// are running on the same host) based on NEXTEST_TEST_GLOBAL_SLOT variable
+/// The port ranges will be reused following nextest logic.
+///
+/// When running without nextest, this will only bump an atomic and eventually
+/// panic when it runs out of port numbers to assign.
+pub fn localhost_port_range_for_tests() -> (u16, u16) {
+    static SLICE: AtomicU16 = AtomicU16::new(0);
```

Comment:
the logic here is a bit convoluted but results in each test validator getting a unique 20-port slice for binding.